src/transformState.js

```diff
@@ -1,11 +1,30 @@
-'use strict';
+"use strict";
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case "addProperties":
+        Object.assign(state, extraData);
+        break;
+
+      case "removeProperties":
+        for (const key of keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case "clear":
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        return "Error: Type Unknown";
```

Comment:
In the 'default' case of your switch statement, you are returning a string. This will end the execution of the function, even if there are more valid actions in the 'actions' array. Instead, consider logging the error or ignoring the action with an invalid type.